### PR TITLE
Dst deduce tests

### PR DESCRIPTION
**Pull request by mmsc2**
Add additional tests to deduce dst

### CODE REVIEW

File: pkg/vm/vm_core_test.go [hunk 1]
```diff
@@ -447,3 +447,54 @@ func TestDeduceDstOpcodeAssertEqWithoutRes(t *testing.T) {
...
+	}
+
+	vm := NewVirtualMachine()
+	vm.RunContext.Fp = memory.NewRelocatable(1, 0)
+
+	result_res := vm.DeduceDst(instruction, nil)
+	mr := memory.NewRelocatable(1, 0)
+	expected_res := memory.NewMaybeRelocatableRelocatable(mr)
+
+	if *result_res != *expected_res {
```
Comment (fmoletta) [inline pkg/vm/vm_core_test.go]:
```suggestion
	result_dst := vm.DeduceDst(instruction, nil)
	mr := memory.NewRelocatable(1, 0)
	expected_dst := memory.NewMaybeRelocatableRelocatable(mr)

	if *result_dst != *expected_dst {
```

File: pkg/vm/vm_core_test.go [hunk 2]
```diff
@@ -447,3 +447,54 @@ func TestDeduceDstOpcodeAssertEqWithoutRes(t *testing.T) {
...
+		t.Error("Different Res value")
```
Comment (fmoletta) [inline pkg/vm/vm_core_test.go]:
```suggestion
		t.Error("Different Dst value")
```

File: pkg/vm/vm_core_test.go [hunk 3]
```diff
@@ -447,3 +447,54 @@ func TestDeduceDstOpcodeAssertEqWithoutRes(t *testing.T) {
...
+		Opcode:   Ret,
+	}
+
+	vm := NewVirtualMachine()
+
+	result_res := vm.DeduceDst(instruction, nil)
+
+	if result_res != nil {
+		t.Error("Different Res value than nil")
+	}
```
Comment (fmoletta) [inline pkg/vm/vm_core_test.go]:
```suggestion
	result_dst := vm.DeduceDst(instruction, nil)

	if result_dst != nil {
		t.Error("Different Dst value than nil")
	}
```

**Review by gabrielbosio — APPROVED** [review]

**Review by gabrielbosio — APPROVED** [review]

**Review by fmoletta — COMMENTED** [review]
Should be moved to vm_test.go

**Review by fmoletta — APPROVED** [review]

**Pull request merged**